src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.ClientCertProvider.cs

```diff
@@ -0,0 +1,95 @@
+using System.Security.Cryptography;
+
+namespace System.Net.Http
+{
+    internal partial class CurlHandler : HttpMessageHandler
+    {
+
+        private sealed class ClientCertProvider : IDisposable
+        {
+            internal readonly string _certificateFile;
+            internal readonly string _password;
+
+            private ClientCertProvider(X509Certificate2Collection certs)
+            {
+                Debug.Assert(certs.Count >0);
+                try
+                {
+                    _password = GeneratePassword();
+                    byte[] pfxData = certs.Export(X509ContentType.Pkcs12, _password);
+                    _certificateFile = Path.GetTempFileName();
+                    File.WriteAllBytes(_certificateFile, pfxData);
+                }
+                catch (Exception e)
+                {
+                    throw new HttpRequestException(SR.net_http_client_execution_error, e);
+                }
+            }
+
+            public void Dispose()
+            {
+                if (_certificateFile == null)
+                {
+                    return;
+                }
+                try
+                {
+                    File.Delete(_certificateFile);
+                }
+                catch (IOException e)
+                {
+                    VerboseTrace("Deleting temporary file:" + e);
+                }
+            }
+
+            internal static ClientCertProvider Create(ClientCertificateOption option, string uriScheme)
+            {
+                if (option == ClientCertificateOption.Manual || !string.Equals(uriScheme, UriSchemeHttps))
+                {
+                    return null;
+                }
+
+                X509Certificate2Collection clientCerts = new X509Certificate2Collection();
+                using (var myStore = new X509Store())
+                {
+                    myStore.Open(OpenFlags.ReadOnly);
+                    X509Certificate2Collection certificates = myStore.Certificates;
+
+                    certificates = certificates.Find(X509FindType.FindByKeyUsage, X509KeyUsageFlags.DigitalSignature, true);
+                    certificates = certificates.Find(X509FindType.FindByApplicationPolicy, ClientAuthenticationOID, true);
+                    foreach (X509Certificate2 cert in certificates)
+                    {
+                        if (cert.HasPrivateKey)
+                        {
+                            clientCerts.Add(cert);
+                        }
+                    }
+                }
+
+                if (clientCerts.Count == 0)
+                {
+                    VerboseTrace("No Client Certs found");
+                    return null;
+                }
+
+                return new ClientCertProvider(clientCerts);
+            }
+
+            private static string GeneratePassword()
+            {
+                byte[] bytes = new byte[52];
+                using (var prng = RandomNumberGenerator.Create())
```

Comment:
The PRNG is the type named `Random`.  `RandomNumberGenerator` is the cryptographically secure random number generator.  `rng` is good enough, or `csrng` if you feel fancy :).